File: serveur/app/models/actual_workout_exercise.py
from sqlalchemy import Column, Integer, ForeignKey, String, Boolean, DateTime, func
from app.database import Base

import uuid

class ActualWorkoutExercise(Base):
    __tablename__ = "actual_workout_exercises"

    id = Column(Integer, primary_key=True, index=True)

    actual_workout_uuid = Column(String, ForeignKey("actual_workouts.uuid", ondelete="CASCADE"), nullable=False, index=True)
    exercise_uuid = Column(String, ForeignKey("exercises.uuid", ondelete="CASCADE"), nullable=False, index=True)

    sets = Column(Integer, nullable=False, default=0)               # Number of sets for the exercise
    reps = Column(String, nullable=False, default="0-1")            # Number of repetitions for the exercise
    phase = Column(String, nullable=False, default="TRAINING", server_default="TRAINING")  # e.g., "WARMUP", "TRAINING", "POST_TRAINING"
    status = Column(String, nullable=False, default="NOT_STARTED")  # e.g., "NOT_STARTED", "IN_PROGRESS", "DONE", "SKIPPED"
    order = Column(Integer, nullable=False, default=0)              # Order of the exercise in the workout
    added_manually = Column(Boolean, default=False)                 # Indicates if the exercise was added manually by the user

    updated_at = Column(DateTime(timezone=True), nullable=True)

    uuid = Column(String, default=lambda: str(uuid.uuid4()), nullable=False, unique=True)

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/ActualWorkoutExercise.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.ForeignKey
import androidx.room.Relation
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import com.google.gson.annotations.SerializedName
import java.util.UUID

@Entity(
    tableName = "actual_workout_exercises",
    indices = [
        androidx.room.Index(value = ["uuid"], unique = true),
        androidx.room.Index(value = ["actual_workout_uuid"]),
        androidx.room.Index(value = ["exercise_uuid"]),
    ],
    foreignKeys = [
        ForeignKey(
            entity = ActualWorkout::class,
            parentColumns = ["uuid"],
            childColumns = ["actual_workout_uuid"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["uuid"],
            childColumns = ["exercise_uuid"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class ActualWorkoutExercise(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "actual_workout_uuid") val actualWorkoutUUID: String,
    @ColumnInfo(name = "exercise_uuid") val exerciseUUID: String,
    @ColumnInfo(defaultValue = "0") val sets: Int = 0,
    @ColumnInfo(defaultValue = "0-1") val reps: String,
    val phase: String,
    val status: String,
    val order: Int,
    @ColumnInfo(name = "added_manually") val addedManually: Boolean = false,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)